### ISSUE TITLE
bernoulli counter

### ISSUE THREAD

**Issue by joopvdl** [phdsg/PvC#10]
So I was going to mention this in the now closed issue that led to bernoulli switch.
What about a bernoulli counter? Imagine Martin Lueder's counter, but with a random factor to it.
The counter knob will set a minimum, but the dice will determinate whether all incoming triggers will be counted. A knob will set the probability.
This could either be a switch or a gate, just like the bernoulli switch.

**Comment by joopvdl** [comment]
btw can't figure out how to label this as feature request

**Comment by phdsg** [comment]
ok, sounds interesting.
have you set this up with modules in a patch?
i'm not sure if i understand exactly what you are trying to achieve...

yeah. i think only the repo owner can label stuff.

**Comment by joopvdl** [comment]
I like randomness. I like structure.
I like counter, because you can tell it to wait, let's say, 8 pulses, then
a gate opens. But I would like a module that I can tell: wait at least 8
pulses, maybe more, you decide. That's where the coin decides. So it will
wait at least 8 pulses, maybe 10, maybe 20, depending on the probability
knob.
I would use it the same way I would use yesterday's switch.
I am working on a patch where I have voices come in after some time. Now,
that time is fixed, but I would like some randomness.
Does that make it more clear?


2018-02-15 10:44 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
so, basically just a bernoulli gate module in front of a counter module like this:
![rainbucket](https://user-images.githubusercontent.com/33329578/36251336-2cbea370-1241-11e8-8c59-754839ed03f2.png)

**Comment by joopvdl** [comment]
yes, so maybe that wouldn't need to be a separate module even )

2018-02-15 11:14 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
might (re)look into this at a later time.
right now i'm trying to get everything ready to deploy 0.5.8.
and then i'll have to prepare the 0.5 -> 0.6 transition.
if time allows maybe some work on the AorBtoOut version of the switch.

**Comment by joopvdl** [comment]
I came up with this idea when I was lying awake last night.
Seemed like a great idea at that time, didn't realise it was so easy to
accomplish.

If you need any help or feedback, please let me know.

Best, Joop

2018-02-15 13:37 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
it's not a bad idea.
and it's also not too hard to frankenstein two modules together into one.
been doing that a lot. most of the time because certain module combinations that i often use take so much time to be patched up and then fill two rows in the rack.
anyways... right now, i'd rather know if the stuff we did yesterday works as expected, doesn't crash, etc.

**Comment by joopvdl** [comment]
going to toy with the bernouilli switch now.
As for he frankenstein plug: usually, i would combine a bernoulli, a switch
and a slew limiter. Yesterday we combined the first two, but perhaps a slew
limiter added would be a nice addition.
You'll hear from me again tonight.

2018-02-15 19:47 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
boy, i see a lot of overlap on module preferences :)
here's a preview of the next commit:
![bernswitches](https://user-images.githubusercontent.com/33329578/36278890-81241784-1295-11e8-8363-36bb432de922.png)

**Comment by joopvdl** [comment]
Yes, seems we have something in common :)
I've been toying around with the modules tonight, no crashes, working as
expected, apart from the probability knob, I would expect it to be the
other way around.



2018-02-15 21:18 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
yeah that part would be easier on a 4HP panel :)
my reasoning was: left = LO / right = HI  
but i can live with either way, changing with next commit.

question: have you ever used the cv input on the audible bernoulli?

**Comment by joopvdl** [comment]
No, never used that.

My reasoning: left=upper outputs
right=lower outputs


2018-02-15 22:43 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
guessed so. i only used them with gate signals to make sure it goes 100% B on certain events.
up until yesterday. found out that the weird behavior i got when using it with non binary signals
was a scaling issue which i copied from the audible. fixed now. upstream too.

yeah i'm not gonna argue over that up/down left/right thing.

**Comment by joopvdl** [comment]
no arguing needed, no, but it's interesting to note we have different
associations. Probably different backgrounds.


2018-02-15 23:05 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
yeah, my brain is wired a little bit weird when it comes to left/right stuff: i often say left when i mean to say right. in this case i just intuitively went left=lo, right=high.
no right or wrong here tho. just a matter of learned taste and habits as you said.

**Comment by phdsg** [comment]
gonna rename this issue and leave it open for general ideas chitchat.
feel free to open specific issues if you find stuff that needs attention in any of the modules.

**Comment by joopvdl** [comment]
perhaps you could name them A and B, just like the name of the module would
suggest.
To me high and low would suggest +10V and 0V, I find that a bit confusing.
You could put a little A and B on the sides of the knob, that makes it even
extra clear.

One more question: why did you change the size of the comparator? I quite
liked it the way it was.

2018-02-15 23:20 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
agree, renamed and inverted knob assignment, but i don't like the 2HP layout.
trying some 4HP variants...
which will also make the assignment and routing more intuitive.

compair: different reasons...main ones being space and visual consistency with the rest of the collection.

**Comment by phdsg** [comment]
opinions on the gate input?
i found it a bit confusing to use in combination with the flip, so i'll probably change that.
either to a one input trigger to reset to default (which was in you request iirc)
or maybe two trigger inputs to route A or B directly (as in shutit).

**Comment by joopvdl** [comment]
I think I agree about the gate input.
A reset to default function is really necessary, but that can also be
achieved by the route to A or B directly function. And that offers some
interesting options as well, so maybe a good idea.
As to the compair and 4hp subject, I think it is very important that the
module is as intuitive as possible, or else people will ignore it. So a
well thought out layout with proper labelling might be a big help in this,
even if it means bigger modules. After all 4hp is still not very big.



2018-02-16 15:24 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
so, this is how they look now...
![bernswitches2](https://user-images.githubusercontent.com/33329578/36317562-b382ce70-133d-11e8-99ab-d55ec3b74ad1.png)
gate gone, direct triggers in, trigger outs added, yet unfinished toss mode switch. that's it so far.
gonna add few buttons to have manual control as well.

compair: agreed on intuitive layouting. the new layout didn't change that much tho and unless there's something really confusing now that wasn't in the big version i'll probably stay with the 4hp version.

**Comment by joopvdl** [comment]
nice!
In de documentation maybe  it would be wise to point out the reset function
that's a bit hidden now.

**Comment by phdsg** [comment]
it don't think of it as a reset function. more of a direct setter for each route (A just being the default).
reminds me that i have to make em save their state...

**Comment by phdsg** [comment]
sort of finished, i guess:
![bernswitches2](https://user-images.githubusercontent.com/33329578/36324665-8a79a946-1355-11e8-8ded-bfdd489c42d2.png)
should now pretty much cover all the things you can do with the regular bernoulli, plus the audio-path.
toss mode is working and sort of the answer to the question i raised before about the effect of the coin toss.

**Comment by joopvdl** [comment]
respect!
looking very nice indeed. Going to use this one a lot!

2018-02-16 20:25 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
here's a new build from current master.
[PvC-0.5.8-win.zip](https://github.com/phdsg/PvC/files/1732667/PvC-0.5.8-win.zip)

**Comment by phdsg** [comment]
offtopic: saw your question on the FB.  
https://www.youtube.com/watch?v=1CoyvGY74PE

**Comment by phdsg** [comment]
ah damn, forgot to mention that the new version is probably gonna break any patch you used them in.
port ids have changed and some other stuff.

**Comment by joopvdl** [comment]
thanks. That's sort of exactly what I was looking for. And, of course,
going to toy around with the modules, too!

2018-02-16 21:15 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
little update on the interface side of things:
panel buttons are now implemented.
![bernswitches3](https://user-images.githubusercontent.com/33329578/36342014-f2fd8e0e-13f7-11e8-9c82-55f9f3b9e045.png)

**Comment by joopvdl** [comment]
wow

2018-02-17 15:36 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
i think i found a name for the pair: "Heads" & "Tails"
because AoRbTObla... meh, didn't like.
if u got a nice name suggestion, shoot.

oh and i pushed some more pixels...
![bernswitches4](https://user-images.githubusercontent.com/33329578/36345813-6194e2c4-1432-11e8-8ae3-011bc779ed11.png)

**Comment by joopvdl** [comment]
your graphics are getting better by the minute :)
Heads and tails is nice.
"Coin" could also work
Or random switch, that's not really sexy but it makes the module easier to
find.
I saw you already released yesterday's version.



2018-02-17 22:35 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
thx. been pushing pixels for while so i know it works, the difficult part is deciding, and coming up with
something that i can use across the collection...
it's more of a pre-release, that's why didn't push it to the plugin manager. there might be patch breaking changes in the next releases because there are many new modules that still are a bit of a work in progress.
i can roll a new binary if you're interested in trying the changes.

**Comment by joopvdl** [comment]
oh yes I would like that.

**Comment by phdsg** [comment]
[PvC-0.5.dev-win.zip](https://github.com/phdsg/PvC/files/1734233/PvC-0.5.dev-win.zip)

**Comment by phdsg** [comment]
ShutIt face lift:
![shutitnew](https://user-images.githubusercontent.com/33329578/36350573-ffdcf4ae-149a-11e8-8313-c5a5affee8d8.PNG)

**Comment by joopvdl** [comment]
btw have you dropped oomph?

2018-02-18 11:01 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
don't tell me i have to resurrect oomph. you're using that? :)

**Comment by joopvdl** [comment]
err, no, actually not :)
Just noticed it was missing..

2018-02-18 16:29 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
phew. i didn't really us it either that's why it's gone.
i might give the whole distortion thing another try some day when i know more about programming dsp, but that's a long road ahead.

**Comment by phdsg** [comment]
uploaded a little bit of a "manual" https://phdsg.github.io/PvC/chance_sw.html
let me know if i missed something critical.

**Comment by joopvdl** [comment]
I feel honored :)
On the facebook some people are reporting problems with reading the text on
the modules because of the color combinations.
Don't know if you've seen that

2018-02-19 0:30 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
yeah, saw that. thx for your kind words. been exiled from fb, so i can only read.
(also have my issues with fb as a community platform, but that's another story.)
as you pointed out correctly, github is the place...
trying to update all the panels and docs for 0.6, but there's only 24h a day :)

![image](https://user-images.githubusercontent.com/33329578/36369569-faecd29a-155b-11e8-82fd-5bc58cf0f9f9.png)

**Comment by joopvdl** [comment]
yeah, facebook, hate it but like to keep up-to-date as to rack.
I doubt whether anyone will make it to the github, though.
exiled sounds serious. Is it?

2018-02-19 10:02 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
well, i had quit fb a long time ago.
only to log on the rack group i reactivated an old account i had used for promo stuff. 
then some day i got reported to the name police and now fb wants to see my real id.
so. nothing serious. just don't wanna jump through that hoop.

**Comment by joopvdl** [comment]
I understand. Couldn't you just make a new account? Well, it's not really
necessary, I guess.
Speaking of names, I've been looking around but I have no idea what your
name is.....

2018-02-19 10:37 GMT+01:00 phdsg vn chfhn <notifications@github.com>:

**Comment by phdsg** [comment]
phdsg is an abbreviation of phadsge which is a mangled version of fatzke which was my nickname back in the counter strike days. my name on fb was phdsg gsdhp i think :) yeah, that one obviously doesn't fit fb criteria for a "real" name. as for real names on the internet. i only use mine where it's absolutely necessary, don't like feeding the commercial data collectors.

**Comment by phdsg** [comment]
Just a bit of an update amidst all the v0.6 updating/testing stuff:

![pre06](https://user-images.githubusercontent.com/33329578/36732985-90c3be86-1bcf-11e8-8331-e64b7bd14649.PNG)

All but 2 modules (button, logic gates) reskinned, minor stuff might still change (port colors for example).   
Old SumIt replaced with a "precision adder"-inspired module.
Will probably add a new module that covers what i used the old SumIt for, but'll also have a few more things i want.  
not sure tho if i find the time to port changes back to 0.5 before 0.6 hits.

**Comment by NRG70** [comment]
oh wowww man! if you ever find the time and inspiration to finish those , 
who knows in time with release of Rack 1.0 
I'lld be happy over the moon to use those!
Take care :)

**Comment by joopvdl** [comment]
Yes, they are still in use, very much so!
It's a pity you removed the manual from the github, because I tend to
advertise some of these tools, but cannot point people towards a manual.
I very much hope you will (at least) find some time to get these great
plugins survive the Rack 1.0 transition.

Joop


Op do 17 jan. 2019 om 13:57 schreef NRG70 <notifications@github.com>:

**Comment by NRG70** [comment]
Hi :) 
could you please make your modules work in upcoming Rack 1.0 release?

Maybe if you have time for it, include those little wonderfull goodies you mentioned above: 2hp modules (including ryo logic clones).

i would be so grateful for that!
